### PR TITLE
HiveMQ replacing VerneMQ

### PR DESCRIPTION
**Pull request by Sphingobium**
Ready for review :)

### CODE REVIEW

**Comment by Scarjit** [comment]
Gitguardian will fail, cause i pushed ~10 private keys :D

**Comment by Scarjit** [comment]
@JeremyTheocharis Let's not merge this, without Documentation updates

File: golang/cmd/sensorconnect/main.go [hunk 1]
```diff
@@ -133,7 +134,7 @@ func main() {
 		if subTwentyMs {
 			zap.S().Warnf("Going under 20MS is not recommendet with IFM IO-Link Masters")
 		} else {
-			panic("LOWER_POLLING_TIME under 20 can IFM IO-Link Master failures, set SUB_TWENTY_MS to 1 to continue !")
+			zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/main.go:137]: really want to remove the concrete error message here?

File: golang/cmd/sensorconnect/main.go [hunk 2]
```diff
@@ -188,7 +189,7 @@ func main() {
 	}
 
 	if deviceFinderTimeoutInS > deviceFinderFrequencyInS {
-		panic("DEVICE_FINDER_TIMEOUT_SEC should never be greater then DEVICE_FINDER_TIME_SEC")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/main.go:192]: really want to remove the concrete error message here?

File: golang/internal/kafka.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func CloseKafka() {
 func CloseKafkaTopicProbeConsumer() {
 	err := KafkaTopicProbeConsumer.Close()
 	if err != nil {
-		panic("Failed do close KafkaTopicProbeConsumer client !")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/internal/kafka.go:77]: really want to remove the concrete error message here?

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]
Please adjust the documentation (or delegate to @Sphingobium) incl.
- adjusting the Helm Chart configuration https://learn.umh.app/docs/core/helmchart/
- all mentions of VerneMQ should now include that it is deprecated, and that we now use HiveMQ
- new tutorial on setting up PKI infrastructure (or adjust existing one) https://learn.umh.app/guides/advanced/production/setup-pki-infrastructure/
- add chapter in documentation -> overview -> data integrity and high availability stating that the system is modeled, that data is stored immediately to disk when using MQTT. However, it is not guaranteed that each data point is actually processed (see also our blog article on scalable data processing). additionally put in for high availability, that for HA in HiveMQ either the commercial license needs to be bought or another tools like VerneMQ should be used
- add to all guides using Node-RED and MQTT, that from now on a specific username and password NEEDS to be used.
-

File: golang/cmd/sensorconnect/main.go (hunk 1 above)
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go:137]: No, those where regex errors

**Comment by Scarjit** [comment]
@JeremyTheocharis Also checkout "Setting up the HiveMQ PKI" on notion

**Comment by Scarjit** [comment]
Added to Changelog (0.9.10)

File: deployment/united-manufacturing-hub/values.yaml [hunk 4]
```diff
@@ -457,7 +491,7 @@ nodered:
...
..."g":"ba92812b.1beda","item":"","datatype":"","value":"","name":"","x":340,"y":540,"wires":[["392b9fc.5171e6"]]},{"id":"392b9fc.5171e6","type":"OpcUa-Client","z":"45f7c4f7.dff1cc","d":true,"g":"ba92812b.1beda","endpoint":"","action":"read","deadbandtype":"a","deadbandvalue":1,"time":10,"timeUnit":"s","certificate":"n","localfile":"","localkeyfile":"","securitymode":"None","securitypolicy":"None","name":"","x":520,"y":540,"wires":[[]]},{"id":"69acedbea0a497d9","type":"inject","z":"41c9c6783cbe3070","name":"","props":[{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"ia/factoryinsight/test/test/count","x":210,"y":180,"wires":[["2ec74b8f684fec8a"]]},{"id":"2ec74b8f684fec8a","type":"function","z":"41c9c6783cbe3070","name":"","func":"msg.payload = {\n\"timestamp_ms\": Date.now(),\n\"count\": 10\n}\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":380,"y":180,"wires":[["b5f390e651f93b89"]]},{"id":"b5f390e651f93b89","type":"mqtt out","z":"41c9c6783cbe3070","name":"","topic":"","qos":"","retain":"","respTopic":"","contentType":"","userProps":"","correl":"","expiry":"","broker":"5dff9791.d1b278","x":530,"y":180,"wires":[]}]
```
Comment (JeremyTheocharis) [inline deployment/united-manufacturing-hub/values.yaml:494]: what changed here?

File: golang/cmd/kafka-state-detector/anomalyKafka.go [hunk 5]
```diff
@@ -54,7 +54,7 @@ func CloseAnomalyKafka() {
 	zap.S().Infof("[Anomaly]Closing Kafka Consumer")
 	err := AnomalyKafkaConsumer.Close()
 	if err != nil {
-		panic("Failed do close AnomalyKafkaConsumer client !")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/kafka-state-detector/anomalyKafka.go:57]: don't we want to have the detailed error message here? "Failed do close AnomalyKafkaConsumer client !

File: golang/cmd/kafka-to-blob/main.go [hunk 6]
```diff
@@ -74,27 +74,27 @@ func main() {
 	if internal.EnvIsTrue("KAFKA_USE_SSL") {
 		securityProtocol = "ssl"
 
-		_, err := os.Open("/SSL_certs/tls.key")
+		_, err := os.Open("/SSL_certs/kafka/tls.key")
 		if err != nil {
-			panic("SSL key file not found")
+			zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/kafka-to-blob/main.go:79]: dont we want to have detailed error messages here?

File: golang/cmd/kafka-to-postgresql/highIntegrityKafka.go [hunk 7]
```diff
@@ -46,7 +46,7 @@ func CloseHIKafka() {
 	zap.S().Infof("[HI]Closing Kafka Consumer")
 
 	if err := HIKafkaConsumer.Close(); err != nil {
-		panic("Failed do close HIKafkaConsumer client !")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/kafka-to-postgresql/highIntegrityKafka.go:49]: dont we want toi have detailed error messages here?

File: golang/cmd/kafka-to-postgresql/highThroughputKafka.go [hunk 8]
```diff
@@ -38,7 +38,7 @@ func CloseHTKafka() {
 	zap.S().Infof("[HT]Closing Kafka Consumer")
 
 	if err := HTKafkaConsumer.Close(); err != nil {
-		panic("Failed do close HTKafkaConsumer client !")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/kafka-to-postgresql/highThroughputKafka.go:41]: see above

File: golang/cmd/kafka-to-postgresql/pxProcessValue.go [hunk 9]
```diff
@@ -204,12 +204,12 @@ func writeProcessValueToDatabase(messages []*kafka.Message) (
 
 					// This function can parse any float, int and bool values.
 					// It will discard any other values with a warning.
-					switch v.(type) {
+					switch t := v.(type) {
```
Comment (JeremyTheocharis) [inline golang/cmd/kafka-to-postgresql/pxProcessValue.go:207]: what does this do? why is it in a HiveMQ PR?

File: golang/cmd/mqtt-kafka-bridge/mqtt.go [hunk 10]
```diff
@@ -21,23 +21,40 @@ func newTLSConfig() *tls.Config {
...
+	zap.S().Debugf("Client cert: %v", cert)
+
 	// Just to print out the client certificate..
 	cert.Leaf, err = x509.ParseCertificate(cert.Certificate[0])
 	if err != nil {
-		panic(err)
+		zap.S().Fatalf("Error parsing client certificate: %s", err)
 	}
 
+	skipVerify := os.Getenv("INSECURE_SKIP_VERIFY") == "true"
```
Comment (JeremyTheocharis) [inline golang/cmd/mqtt-kafka-bridge/mqtt.go:56]: variable is not documented yet, also not consistent with mqtt-bridge (there is no variable yet)

File: golang/cmd/sensorconnect/kafka.go [hunk 11]
```diff
@@ -20,23 +20,26 @@ func SendKafkaMessage(kafkaTopicName string, message []byte) {
...
+			"Duplicate message for topic %s, you might want to increase LOWER_POLLING_TIME !",
+			kafkaTopicName)
 		return
 	}
 
 	err := internal.CreateTopicIfNotExists(kafkaTopicName)
 	if err != nil {
 		zap.S().Errorf("Failed to create topic %s", err)
-		panic("Failed to create topic, restarting")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/kafka.go:32]: can we add the original error message?

File: golang/cmd/sensorconnect/kafka.go [hunk 12]
```diff
@@ -53,37 +56,37 @@ func setupKafka(boostrapServer string) (producer *kafka.Producer, adminClient *k
 	if internal.EnvIsTrue("KAFKA_USE_SSL") {
 		securityProtocol = "ssl"
 
-		_, err := os.Open("/SSL_certs/tls.key")
+		_, err := os.Open("/SSL_certs/kafka/tls.key")
 		if err != nil {
-			panic("SSL key file not found")
+			zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/kafka.go:61]: can we add the original error messages

File: golang/cmd/sensorconnect/main.go (hunk 1 above)
Comment (JeremyTheocharis) [reply golang/cmd/sensorconnect/main.go:137]: maybe push your changes? at the moment these are not active yet

File: golang/internal/kafka.go [hunk 13]
```diff
@@ -62,7 +62,7 @@ func SetupKafkaTopicProbeConsumer(configMap kafka.ConfigMap) {
 func CloseKafka() {
 
 	if err := KafkaConsumer.Close(); err != nil {
-		panic("Failed do close KafkaConsumer client !")
+		zap.S().Fatal(err)
```
Comment (JeremyTheocharis) [inline golang/internal/kafka.go:65]: regex error as well?

File: deployment/united-manufacturing-hub/values.yaml [hunk 14]
```diff
@@ -219,25 +253,14 @@ _000_commonConfig:
   # mqttBridge allows connecting multiple United Manufacturing Hub clusters, e.g., the installations on the edge with the central cloud
   mqttBridge:
     enabled: false
-    remoteBrokerURL: "ssl://united-manufacturing-hub-vernemq-local-service.united-manufacturing-hub:8883"
     localSubTopic: "ia/factoryinsight"  # /# will be automatically appended to the string specified here.
     oneWay: true
-    localBrokerSSLEnabled: false
     localPubTopic: "ia"
-    localCertificateName: "LOCAL"
+    remoteBrokerURL: "ssl://united-manufacturing-hub-vernemq-local-service.united-manufacturing-hub:8883"
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml:259]: No

File: golang/cmd/mqtt-kafka-bridge/mqtt.go (hunk 10 above)
Comment (Scarjit) [reply golang/cmd/mqtt-kafka-bridge/mqtt.go:56]: Added documentation in values.yaml

**Review by JeremyTheocharis — APPROVED** [review]

**Review by JeremyTheocharis — APPROVED** [review]

**Review by Scarjit — APPROVED** [review]

**Pull request merged**